server/webhook.go

```diff
@@ -260,6 +260,267 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 		}
 		return c.String(http.StatusOK, strings.Join(createdMessageList, "\n"))
 	})
+	g.POST("/github/:id", func(c echo.Context) error {
+		ctx := c.Request().Context()
+		body, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to read webhook request").SetInternal(err)
+		}
+
+		// This shouldn't happen as we only setup webhook to receive push event, just in case.
+		eventType := github.WebhookType(c.Request().Header.Get("X-GitHub-Event"))
+		if eventType != github.WebhookPush {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Invalid webhook event type, got %s, want %s", eventType, github.WebhookPush))
+		}
+
+		var pushEvent github.WebhookPushEvent
+		if err := json.Unmarshal(body, &pushEvent); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed push event").SetInternal(err)
+		}
+
+		webhookEndpointID := c.Param("id")
+		repo, err := s.store.GetRepository(ctx, &api.RepositoryFind{WebhookEndpointID: &webhookEndpointID})
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to respond webhook event for endpoint: %v", webhookEndpointID)).SetInternal(err)
+		}
+		if repo == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Webhook endpoint not found: %v", webhookEndpointID))
+		}
+
+		if repo.VCS == nil {
+			err := fmt.Errorf("VCS not found for ID: %v", repo.VCSID)
+			return echo.NewHTTPError(http.StatusInternalServerError, err).SetInternal(err)
+		}
+
+		//nolint:misspell
+		// TODO(unknwon): Validate webhook signature, https://linear.app/bbteam/issue/BYT-935
+
+		if pushEvent.Repository.FullName != repo.ExternalID {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Project mismatch, got %s, want %s", pushEvent.Repository.FullName, repo.ExternalID))
+		}
+
+		log.Debug("Processing GitHub webhook push event...",
+			zap.String("project", repo.Project.Name),
+		)
+
+		var createdMessageList []string
+		for _, commit := range pushEvent.Commits {
+			// The Distinct is false if the commit is superseded by a later commit.
+			if !commit.Distinct {
+				continue
+			}
```

Comment:
This addresses the similar problem as in https://linear.app/bbteam/issue/BYT-857/for-vcs-workflow-duplicated-issues-are-created-when-the-commit-is